test/apm_ec2/resources/run_java_application.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+JAVA_TOOL_OPTIONS=" -javaagent:${HOME}/aws-apm-0.3.0.jar" \
```

Comment:
Is there another option than to hardcode the java APM version? Also, if this is a CW agent test that uses the java agent, doesn't it make more sense as an E2E test?

Reply:
The problem here is that ADOT only provides numbered versions, there's no `latest` jar file available, so even if we make it configurable, we still need to come back to this repo and update it.

But your concern makes sense, making it hard coded in the script is not ideal. I would recommend we revisit this after RIV when we have an official place to download the binary.

> doesn't it make more sense as an E2E test?
Sorry, not quite following, it is E2E test, right? We use java agent to generate Pulse attributes so that we can verify that the `awsapmprocessor` in the CW agent translates these attributes correctly.

Reply:
The `amazon-cloudwatch-agent-test` repo is meant for agent integration tests, not "e2e" tests that were setup in the main repo. 

This repo tests the agent by itself by crafting an otlp json request to the agent and expects some metric/trace.

If this test uses the java agent for testing, I'd recommend having this as part of the current e2e tests. @majanjua-amzn  would you agree?

Reply:
Agree with Lisa, although if this accomplishes the goal of the test we can leave it as is for now and come back to it later when we're working on moving the E2E tests to a centralized location.